blogposts/2023-06-07-metalhead-v0.8.md

```diff
@@ -0,0 +1,79 @@
+
+The “mid-level” interface allows users looking for more advanced options to start working with models that offer a little more out of the box, without compromising on user ease. In particular, the ResNet family of models has undergone a *huge* revamp, with support for many new modifications added in [https://github.com/FluxML/Metalhead.jl/pull/174](https://github.com/FluxML/Metalhead.jl/pull/174). These modifications include a large number of recent advancements from papers such as [Bag of Tricks](https://arxiv.org/pdf/1812.01187):
+
+1. Stochastic Depth and DropBlock are now supported by the underlying `resnet` function.
+2. The `resnet` function also now takes an attention-layer argument, which allows us to design SE-ResNets and SE-ResNeXts. The `Layers` module also exports an `efficient_squeeze_excite` function for any users who want a computationally efficient attention function.
+3. The stem of the `resnet` can now be modified to be optionally deeper to match the InceptionResNet-v2 stem. An experimental option from the `timm` library in Python to make the deeper stem have a larger width in the second convolution is also supported since this shows improvements over the regular deeper stem in certain cases.
+4. Three types of downsampling layers are now supported. In addition to the identity and convolutional projections for skip connection downsampling, a max pooling+convolution based downsampling projection is also supported now.
+5. The model supports a `revnorm` option, which places the normalisation layers before the convolutional layers.
+
+The documentation for ResNet in Metalhead goes over these options in detail, and there is also a guide on how to use some of these. Head there to check out even more cool features!
+
+### Reducing code duplication and making the low-level interface more powerful
+
+The “low-level” interface allows engineers and researchers to iterate on model design quickly without thinking too much about the building blocks. In particular, multiple dispatch means that functions that need to be re-written again and again in Python libraries can simply be unified under one function in Julia.
+
+One of the major changes Metalhead v0.8 introduces is the concept of *builders.* These are functions that return closures over the **stage and block indices**. These two numbers together index into a block completely (the stage index represents which stage of the large model is being referred to, and the block index finds the specific block inside the stage). Builders have allowed writing functions that construct blocks purely based on the stage and block indices, abstracting away construction of the layers as well as details like calculations for strides, output feature maps or probabilities for stochastic depth/DropBlock. These builders have been used to rewrite the underlying implementations of the three largest CNN families supported by Metalhead and that are used commonly – ResNets, MobileNets and EfficientNets.
```

Comment:
```suggestion
One of the major changes Metalhead v0.8 introduces is the concept of *builders.* These are functions that return closures over the **stage and block indices**. These two numbers together index into a block completely (the stage index represents which stage of the large model is being referred to, and the block index finds the specific block inside the stage). Builders have allowed writing functions that construct blocks purely based on the stage and block indices, abstracting away construction of the layers as well as details like calculations for strides, output feature maps or probabilities for stochastic depth/DropBlock. These builders have been used to rewrite the underlying implementations of the three largest CNN families supported by Metalhead and that are used commonly – [ResNets](https://fluxml.ai/Metalhead.jl/v0.8/api/resnet/#Block-builders), MobileNets and EfficientNets.
```

Reply:
That page has a misleading title 😅 Here we are taking about the model builders, not the block builders.